sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -0,0 +1,71 @@
+#include <BedrockCommand.h>
+#include <sqlitecluster/SQLiteClusterMessenger.h>
+#include <sqlitecluster/SQLiteNode.h>
+
+SQLiteClusterMessenger::SQLiteClusterMessenger(shared_ptr<SQLiteNode>& node)
+ : _node(node)
+{
+}
+
+bool SQLiteClusterMessenger::sendToLeader(BedrockCommand& command) {
+    string leaderAddress;
+    auto _nodeCopy = atomic_load(&_node);
+    if (_nodeCopy) {
+        for (SQLiteNode::Peer* peer : _nodeCopy->peerList) {
+            if (peer->state == STCPNode::LEADING && !peer->commandAddress.load().empty()) {
+                leaderAddress = peer->commandAddress;
+                break;
+            }
+        }
+    }
+
+    // SParseURI expects a typical http or https scheme.
+    string url = "http://" + leaderAddress;
+    string host, path;
+    if (!SParseURI(url, host, path) || !SHostIsValid(host)) {
+        return false;
+    }
+
+    // Create a new transaction. This can throw if `validate` fails. We explicitly do this *before* creating a socket.
+    Transaction* transaction = new Transaction(*this);
+
+    // I don't trust this not to ever leak currently, but for the moment, this is OK.
+    _transactionCommands[transaction] = make_pair(&command, STimeNow());
```

Comment:
Can you elaborate on the comment? Specifically wondering how it may leak and why it is "ok for the moment"

Reply:
I don't think it leaks in the general case. I wonder if it can leak in some exception case where we never get a response to one of these messages or something. I doubt this would be a serious issue, and could be cleaned up later if discovered.